src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -2104,14 +2110,49 @@ private String launchCommandIn(ArgumentListBuilder args, File workDir, EnvVars e
             }
             int usedTimeout = timeout == null ? TIMEOUT : timeout;
             listener.getLogger().println(" > " + command + TIMEOUT_LOG_PREFIX + usedTimeout);
-            Launcher.ProcStarter p = launcher.launch().cmds(args.toCommandArray()).
-                    envs(freshEnv).stdout(fos).stderr(err);
-            if (workDir != null) p.pwd(workDir);
-            int status = p.start().joinWithTimeout(usedTimeout, TimeUnit.MINUTES, listener);
 
-            String result = fos.toString(Charset.defaultCharset().toString());
+            int status=0;
+            String result = "";
+            String errorString = "";
+            if(ZosCheck() == true) {
+                // Another behavior on z/OS required due to the race condition
+                Launcher.ProcStarter p = launcher.launch().cmds(args.toCommandArray()).
```

Comment:
Please provide more details in this comment on the race condition that is requiring this difference of behavior.  For example:

* What is the race condition?
* Where is it described?
* What makes it unique to z/OS?
* How would we detect at some future time that the race condition no longer exists?

Reply:
It is something internal with IBM Java on z/OS implementation - i didn't get much into details there. It seems Runtime.exec() behaves a little bit different when JVM does transcoding on z/OS. As result if i do us LocalProc implementation with StreamCopyThread - i'm getting garbage in a first few chars and i didn't found a way to overcome that easily.  Seems IBM Java implementation of Runtime.exec() is not that good when it have to deal with transcoding - that's why there is IBM z/OS Java own version of exec() in JZOS library - which do work great https://www.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/com.ibm.java.zsecurity.api.80.doc/com.ibm.jzos/com/ibm/jzos/Exec.html  - but i don't want to add that kind of dependency in the project.